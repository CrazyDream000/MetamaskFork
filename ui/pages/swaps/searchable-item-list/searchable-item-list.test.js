import React from 'react';
import configureMockStore from 'redux-mock-store';

import {
  renderWithProvider,
  createSwapsMockStore,
} from '../../../../test/jest';
import SearchableItemList from '.';

const createProps = (customProps = {}) => {
  return {
    defaultToAll: true,
    listTitle: 'listTitle',
    itemsToSearch: [
      {
        iconUrl: 'iconUrl',
        selected: true,
        primaryLabel: 'primaryLabel',
        secondaryLabel: 'secondaryLabel',
        rightPrimaryLabel: 'rightPrimaryLabel',
        rightSecondaryLabel: 'rightSecondaryLabel',
      },
    ],
    fuseSearchKeys: [
      {
        name: 'name',
        weight: 0.499,
      },
      {
        name: 'symbol',
        weight: 0.499,
      },
      {
        name: 'address',
        weight: 0.002,
      },
    ],
    ...customProps,
  };
};

describe('SearchableItemList', () => {
  it('renders the component with initial props', () => {
    const store = configureMockStore()(createSwapsMockStore());
    const props = createProps();
    const { getByText } = renderWithProvider(
      <SearchableItemList {...props} />,
      store,
    );
    expect(getByText(props.listTitle)).toBeInTheDocument();
    expect(getByText(props.itemsToSearch[0].primaryLabel)).toBeInTheDocument();
    expect(
      getByText(props.itemsToSearch[0].secondaryLabel),
    ).toBeInTheDocument();
    expect(
      getByText(props.itemsToSearch[0].rightPrimaryLabel),
    ).toBeInTheDocument();
    expect(
      getByText(props.itemsToSearch[0].rightSecondaryLabel),
    ).toBeInTheDocument();
    expect(
      document.querySelector('.searchable-item-list__search'),
    ).toMatchSnapshot();
    expect(
      document.querySelector('.searchable-item-list__item'),
    ).toMatchSnapshot();
  });
});
