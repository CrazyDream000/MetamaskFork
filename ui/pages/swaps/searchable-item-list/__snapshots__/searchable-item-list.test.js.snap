// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchableItemList renders the component with initial props 1`] = `
<div
  class="MuiFormControl-root MuiTextField-root searchable-item-list__search MuiFormControl-fullWidth"
  data-testid="search-list-items"
>
  <div
    class="MuiInputBase-root MuiInput-root TextField-inputRoot-12 MuiInputBase-fullWidth MuiInput-fullWidth Mui-focused Mui-focused TextField-inputFocused-11 MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
  >
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionStart"
      style="margin-right: 12px;"
    >
      <img
        alt=""
        height="17"
        src="images/search.svg"
        width="17"
      />
    </div>
    <input
      aria-invalid="false"
      autocomplete="off"
      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
      dir="auto"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`SearchableItemList renders the component with initial props 2`] = `
<div
  class="searchable-item-list__item searchable-item-list__item--selected"
  tabindex="0"
>
  <img
    alt=""
    class="url-icon"
    src="iconUrl"
  />
  <div
    class="searchable-item-list__labels"
  >
    <div
      class="searchable-item-list__item-labels"
    >
      <span
        class="searchable-item-list__primary-label"
      >
        primaryLabel
      </span>
      <span
        class="searchable-item-list__secondary-label"
      >
        secondaryLabel
      </span>
    </div>
    <div
      class="searchable-item-list__right-labels"
    >
      <span
        class="searchable-item-list__right-primary-label"
      >
        rightPrimaryLabel
      </span>
      <span
        class="searchable-item-list__right-secondary-label"
      >
        rightSecondaryLabel
      </span>
    </div>
  </div>
</div>
`;
